src/intervals/construction.jl

```diff
@@ -0,0 +1,237 @@
+"""
+    IntervalArithmetic.default_bound()
+
+Return the default bound for intervals.
+
+Can be redefined to change the default behavior for constructors that do
+not explicitly take the bound type as an argument, like `..`.
+
+Intervals only support bounds that are `AbstractFloat`.
+
+In general, what happens is that if none of the bounds is an `AbstractFloat`
+they get promoted to the default bound.
+Otherwise the constructors promote the two bounds to a common type and use
+that one as bound type.
+
+Example
+=======
+julia> IntervalArithmetic.default_bound()
+Float64
+
+julia> typeof(1..2)  # Both bounds are Int, so they get promoted to the default
+Interval{Float64}
+
+julia> typeof(1..2f0)  # One of the bound is Float32, so the bounds promote to it
+Interval{Float32}
+
+julia> IntervalArithmetic.default_bound() = Float32  # New default
+
+julia> typeof(1..2)  # Both bounds are Int, so they get promoted to the default
+Interval{Float32}
+
+julia> typeof(1..2.0)  # One of the bound is Float64, so the bounds promote to it
+Interval{Float64}
+"""
+default_bound() = Float64
+
+@inline _normalisezero(a::Real) = ifelse(iszero(a) && signbit(a), copysign(a, 1), a)
+
+"""
+    Interval
+
+An interval for guaranteed computation.
+
+For the most part, it can be used as a drop-in replacement for real numbers.
+
+The computation is guaranteed in the sense that the image of the input
+interval is guaranteed to lie inside the returned interval.
+
+The validity of the interval is *not* tested by this bare constructor.
+For a constructor that complies with the requirement of a constructor
+according to the IEEE Std 1788-2015, use one of the other convenience
+constructors:
+- `checked_interval(a, b)` Explicitly checked interval.
```

Comment:
why not using `interval` instead of `checked_interval` as it currently is? It would be less breaking and less typing. Also, I don't think we need both `checked_interval` and `..` if they behave the same.

Reply:
I always found `interval` vs `Interval` rather confusing, but don't have a particularly strong opinion here either. Especially if we keep `..` and `Interval` to nearly synonymous (in the sense we don't do anything smart to widen the bounds).

Then again, the advantage if we keep `interval` only is that I can replace all occurrence in one go with `Ctrl + Maj + F` :).